src/System.Private.CoreLib/shared/System/String.cs

```diff
@@ -655,6 +651,30 @@ internal static unsafe int wcslen(char* ptr)
             return count;
         }
 
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static unsafe int strlen(byte* ptr)
+        {
+            if (ptr == null)
+            {
+                return 0;
+            }
+
+            // IndexOf processes memory in aligned chunks, and thus it won't crash even if it accesses memory beyond the null terminator.
+            int length = SpanHelpers.IndexOf(ref *ptr, (byte)'\0', int.MaxValue);
```

Comment:
This has corner-case behavior difference when the string is `int.MaxValue` long. The existing strlen would work, but it is going to throw now instead. It should be ok: It is either not possible to create string this long; or the surrounding code would throw anyway. In general, it is very unlikely that somebody is going to have `int.MaxValue` and expect it to work.